marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/entities/entity-search-lib.sjs

```diff
@@ -55,19 +60,56 @@ function addPropertiesToSearchResponse(entityName, searchResponse, propertiesToD
       console.log(`Unable to obtain entity instance from document with URI '${result.uri}'; will not add entity properties to its search result`);
     }
 
+    let entityDef = entityModel.definitions[entityName];
     if (instance != null) {
       const entityInstance = instance[entityName];
       if (!entityInstance) {
         console.log(`Unable to obtain entity instance from document with URI '${result.uri}' and entity name '${entityName}'; will not add entity properties to its search result`);
       } else {
         selectedPropertyMetadata.forEach(parentProperty => {
-          result.entityProperties.push(getPropertyValues(parentProperty, entityInstance));
-        });
+            result.entityProperties.push(getPropertyValues(parentProperty, entityInstance));
+
+            if (entityDef.hasOwnProperty("primaryKey")) {
```

Comment:
I was able to comment out most of the new code here and the tests all still pass. I did this:

```
            if (entityDef.hasOwnProperty("primaryKey")) {
              let primaryKey = entityDef.primaryKey;
              if (entityInstance.hasOwnProperty((primaryKey))) {
                  result.primaryKey.propertyPath = primaryKey;
                  result.primaryKey.propertyValue = entityInstance[primaryKey];
              } else { // no primaryKey in entityInstance, so use uri
                result.primaryKey.propertyPath = "uri";
                result.primaryKey.propertyValue = result.uri;
              }
            }
```

So I'm not clear on why all the other logic is needed. If it's for the All Entities view, I don't think we care about selected properties there, right? The user shouldn't be able to do that for that feature. In that context, we know the identifier should always be first. 

Also, in the above logic, it's not handling the case where there's no primaryKey property defined, right? I think what we really want is a getPrimaryValue(entityInstance, entityDefinition) function that will return an object containing propertyPath/propertyValue, or it will return null. If it returns null, then use URI.